core/src/main/java/com/radixdlt/keys/SelfValidatorInfoModule.java

```diff
@@ -68,15 +68,17 @@
 import com.google.inject.Provides;
 import com.radixdlt.consensus.bft.BFTValidatorId;
 import com.radixdlt.consensus.bft.Self;
+import com.radixdlt.consensus.bft.SelfValidatorInfo;
 import com.radixdlt.crypto.ECDSASecp256k1PublicKey;
 import com.radixdlt.rev2.ComponentAddress;
 import java.util.Optional;
 
-public final class BFTValidatorIdModule extends AbstractModule {
+public final class SelfValidatorInfoModule extends AbstractModule {
   @Provides
   @Self
-  public BFTValidatorId validatorId(
+  public SelfValidatorInfo selfValidatorInfo(
       @Self Optional<ComponentAddress> validatorAddress, @Self ECDSASecp256k1PublicKey key) {
-    return BFTValidatorId.create(validatorAddress.orElse(null), key);
+    return new SelfValidatorInfo(
```

Comment:
It feels a little weird to me this takes a @Self Optional<ComponentAddress> rather than computing that here?

Like it's weird to me that `@Self Optional<ComponentAddress>` is bound in the DI at all? SelfValidatorInfo feels like a nicer object to rely on :)

Reply:
:+1: